tron/mesos.py

```diff
@@ -0,0 +1,272 @@
+import json
+import logging
+from urllib.parse import urlparse
+
+import requests
+from task_processing.runners.subscription import Subscription
+from task_processing.task_processor import TaskProcessor
+from twisted.internet.defer import logError
+
+from tron.actioncommand import ActionCommand
+from tron.utils.dicts import get_deep
+from tron.utils.queue import PyDeferredQueue
+
+TASK_LOG_FORMAT = '%(asctime)s %(name)s %(levelname)s %(message)s'
+# TODO: put in configs
+MESOS_MASTER_PORT = 5050
+DOCKERCFG_LOCATION = "file:///root/.dockercfg"
+MESOS_SECRET = ''
+MESOS_ROLE = '*'
+OFFER_TIMEOUT = 300
+
+log = logging.getLogger(__name__)
+_processor = None
+frameworks = {}  # TODO: improve
+
+
+def get_mesos_processor():
```

Comment:
any need to have this as a seperate function, rather than just creating it in the constructor of the MesosCluster object?

Reply:
The idea was that the processor might be expensive to create, but after reading the TaskProcessor code, seems fine. Also, only creating it once per cluster seems okay anyway. I can get rid of this.